libfwupd/fwupd-client.c

```diff
@@ -121,13 +123,116 @@ fwupd_client_curl_helper_free (FwupdCurlHelper *helper)
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(FwupdCurlHelper, fwupd_client_curl_helper_free)
 #endif
 
+typedef struct {
+	FwupdClient	*self;
+	gchar		*property_name;
+	guint		 signal_id;
+	FwupdDevice	*device;
+} FwupdClientContextHelper;
+
+static void
+fwupd_client_context_helper_free (FwupdClientContextHelper *helper)
+{
+	if (helper->device != NULL)
+		g_object_unref (helper->device);
+	g_object_unref (helper->self);
+	g_free (helper->property_name);
+	g_free (helper);
+}
+
+static gboolean
+fwupd_client_context_idle_cb (gpointer user_data)
+{
+	FwupdClientContextHelper *helper = (FwupdClientContextHelper *) user_data;
+	FwupdClient *self = FWUPD_CLIENT (helper->self);
+	FwupdClientPrivate *priv = GET_PRIVATE (self);
+	g_autoptr(GMutexLocker) locker = NULL;
+
+	/* property */
+	if (helper->property_name != NULL)
+		g_object_notify (G_OBJECT (self), helper->property_name);
+
+	/* legacy signal name */
+	if (g_strcmp0 (helper->property_name, "status") == 0)
+		g_signal_emit (self, signals[SIGNAL_STATUS_CHANGED], 0, priv->status);
```

Comment:
Is reading `priv->status` here thread-safe?

Reply:
Do you think it should be stored in the helper?

Reply:
The status is in reference to the daemon and shared state, rather than the transaction. All threads should have the same value and I think using a mutex for this legacy thing isn't required.

Reply:
“All threads should have the same value” :rotating_light: Citation needed! :rotating_light: 

Storing it in the helper (so the value which is emitted in the signal is the value from when the signal was scheduled to be emitted) makes sense and should be simpler than reasoning about the thread-safety of `priv->status`.

Reply:
Are you not going to store `priv->status` in the helper?